hack/generator/pkg/codegen/pipeline_mark_storage_version.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+	"sort"
+
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+)
+
+// exportPackages creates a PipelineStage to export our generated code as a set of packages
+func markStorageVersion() PipelineStage {
+	return MakePipelineStage(
+		"markStorageVersion",
+		"Marking the latest version of type as the storage version",
+		func(ctx context.Context, types astmodel.Types) (astmodel.Types, error) {
+			updatedDefs, err := MarkLatestResourceVersionsForStorage(types)
+			if err != nil {
+				return nil, errors.Wrapf(err, "unable to mark latest resource versions for as storage versions")
+			}
+
+			return updatedDefs, nil
+		})
+}
+
+// MarkLatestResourceVersionsForStorage marks the latest version of each resource as the storage version
+func MarkLatestResourceVersionsForStorage(types astmodel.Types) (astmodel.Types, error) {
+
+	result := make(astmodel.Types)
+	resourceLookup, err := groupResourcesByVersion(types)
+	if err != nil {
+		return nil, err
```

Comment:
Use `errors.Warp()` to give more information about the error.

Reply:
There's not really much to add here other than "calling function X failed" - and the function errors can/should already be saying that so I don't think there's a need to wrap here?

Reply:
Fair enough.